lib/livebook/teams/deployment_group.ex

```diff
@@ -50,10 +50,11 @@ defmodule Livebook.Teams.DeploymentGroup do
         end
       end)
 
-    if get_field(changeset, :zta_provider) do
-      validate_required(changeset, [:zta_key])
-    else
+    if get_field(changeset, :zta_provider) in [nil, :livebook_teams] do
       changeset
+    else
+      # A legacy :zta_provider value, which also requires :zta_key
+      validate_required(changeset, [:zta_key])
     end
```

Comment:
We can just skip this, since we should never have the zta_key on this side of the app. We can probably remove zta_key everywhere here.

Reply:
We should also remove it from the Dockerfile generation, right?

Reply:
Yes!